src/Entity/OgMembership.php

```diff
@@ -320,6 +320,22 @@ public function preSave(EntityStorageInterface $storage) {
       throw new \LogicException('OG membership can not be created for an empty or anonymous user.');
     }
 
+    if (!$this->get('entity_id')->value) {
+      // Group was not set.
+      throw new \LogicException('Membership cannot be set for an empty or an unsaved group.');
```

Comment:
There is a way to set the validation flag via `\Drupal\Core\Entity\ContentEntityBase::setValidationRequired` in it would be our responsibility to do so. As far as I can see, currently in core it's being enabled for forms only in `\Drupal\Core\Entity\ContentEntityForm::buildEntity`.

As we need to make sure there isn't wrong data, we'll have to call `$this->setValidationRequired(TRUE)` ourself from `preSave`